lisa/sut_orchestrator/azure/common.py

```diff
@@ -1036,6 +1048,40 @@ def delete_file_share(
     share_service_client.delete_share(file_share_name)
 
 
+def save_console_log(
+    resource_group_name: str,
+    vm_name: str,
+    platform: "AzurePlatform",
+    log: Logger,
+    saved_path: Optional[Path],
+    screenshot_file_name: str = "serial_console",
+) -> bytes:
+    diagnostic_data = _get_diagnostic_data(platform, resource_group_name, vm_name)
+    if saved_path:
+        screenshot_raw_name = saved_path.joinpath(f"{screenshot_file_name}.bmp")
+        screenshot_name = saved_path.joinpath(f"{screenshot_file_name}.png")
+        screenshot_response = requests.get(diagnostic_data.console_screenshot_blob_uri)
+        with open(screenshot_raw_name, mode="wb") as f:
+            f.write(screenshot_response.content)
+        try:
+            with Image.open(screenshot_raw_name) as image:
+                image.save(screenshot_name, "PNG", optimize=True)
+        except UnidentifiedImageError:
+            log.debug(
+                "The screenshot is not generated. "
+                "The reason may be the VM is not started."
+            )
+        unlink(screenshot_raw_name)
+
+    log_response = requests.get(diagnostic_data.serial_console_log_blob_uri)
+    if 404 == log_response.status_code:
```

Comment:
Convention is to have the value you are checking on the left
```suggestion
    if log_response.status_code == 404:
```

Reply:
The `404 == ...` pattern is coming from C. It can avoid mistyped like `404 = ...`. It will cause a compiler error.

Reply:
This not C. `if log_response = 404:` raises a syntax error in Python, so you don't need to do things like this. The Zen of Python says "Readability matter". I think this falls under that.

Reply:
updated